### DISCUSSION TITLE
RegistrationHash: double bond stereochemistry

### DISCUSSION THREAD

**Discussion by kienerj** [rdkit/rdkit#6318] [Q&A]
I'm playing around with the new registration hash and not sure if I'm missing something or if this is a real issue.

Goal:
- tautomer insensitive hash (different tautomer = same hash) 
- stereo sensitive including double bond stereochemisty
- works with molecules inchi fails

last point is relevant as standard inchi already works "good enough" mostly except for when no inchi can be created from a valid rdkit molecule.

However it seems that canonical SMILES is the only layer that takes double bond stereo into account and removing that layer then gives the same hash for different double bond stereo chemistry.

There is no possibility to make the hash tautomer insensitive but double-bond stereo sensitive or what am I missing?

Toy example:

![image](https://user-images.githubusercontent.com/1567309/234558304-cced9fed-3146-4952-9131-afb5b7737ca0.png)

**Comment by greglandrum** [comment]
@kienerj, the tautomer hash removes stereo from any bonds which it considers to be part of a tautomeric region, and there's no option to turn this off.
In your case this spans the conjugated double bonds connecting the N to the O:
```
In [2]: m = Chem.MolFromSmiles('CC[C@H](N)/C=C/C=C/CO')

In [3]: from rdkit.Chem import rdMolHash

In [4]: rdMolHash.MolHash(m,rdMolHash.HashFunction.HetAtomTautomer)
Out[4]: 'CC[C@H]([N])[CH][CH][CH][CH]C[O]_3_0'
```

**Reply by kienerj** [reply]
A better example:

![image](https://user-images.githubusercontent.com/1567309/234637699-6efa819e-4e7e-48d3-bf6f-f0ddc601dda9.png)

I'm always thinking about a larger collection of molecules which might contain relevant tautomers that should result in the same hash. Therefore I would need to have tautomer insensitive turned on while preserving double bond stereo for this above example where tautomerism doesn't matter.

**Comment by kienerj** [comment]
This seems to be solved in 2023.03 release but one has to explicitly use the new tautomer hash by using the argument `enable_tautomer_hash_v2=True`:

    RegistrationHash.GetMolLayers(m, enable_tautomer_hash_v2=True)

For above example the tautomer hash then is different:

![image](https://github.com/rdkit/rdkit/assets/1567309/a333fbf3-e289-4144-9642-9d1645b44f91)
